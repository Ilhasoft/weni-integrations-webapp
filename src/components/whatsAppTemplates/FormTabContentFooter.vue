<template>
  <div class="form-tab-content-footer">
    <unnnic-tag
      class="form-tab-content-footer__tag"
      type="default"
      :text="$t('WhatsApp.templates.form_field.footer_text')"
      scheme="neutral-darkest"
    />

    <unnnic-text-area
      class="form-tab-content-footer__input"
      :disabled="disableInputs"
      :maxLength="60"
      :value="templateTranslationCurrentForm.footer || ''"
      @input="$emit('input-change', { fieldName: 'footer', fieldValue: $event })"
    />
  </div>
</template>

<script>
  import { mapGetters } from 'vuex';

  export default {
    name: 'FormTabContentFooter',
    props: {
      disableInputs: {
        type: Boolean,
        default: false,
      },
    },
    computed: {
      ...mapGetters('WhatsApp', ['templateTranslationCurrentForm']),
    },
  };
</script>

<style lang="scss" scoped>
  .form-tab-content-footer {
    display: flex;
    flex-direction: column;

    &__tag {
      width: fit-content;
      margin-bottom: $unnnic-spacing-stack-sm;
    }

    &__input {
      ::v-deep textarea {
        resize: none;
      }

      ::v-deep .helper {
        margin-top: $unnnic-spacing-stack-xs;
      }
    }
  }
</style>
