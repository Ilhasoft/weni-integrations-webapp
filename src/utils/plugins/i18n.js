import Vue from 'vue';
import VueI18n from 'vue-i18n';

Vue.use(VueI18n);

import en from '../../locales/en.json';
import pt_br from '../../locales/pt_br.json';
import es_es from '../../locales/es_es.json';

const languages = {
  'en-us': en,
  'pt-br': pt_br,
  es: es_es,
};

const messages = Object.assign(languages);

const i18n = new VueI18n({
  locale: 'en-us',
  fallbackLocale: 'en-us',
  messages,
});

export default i18n;
