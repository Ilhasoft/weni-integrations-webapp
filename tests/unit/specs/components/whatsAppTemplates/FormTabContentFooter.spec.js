import Vuex from 'vuex';
import { mount, createLocalVue } from '@vue/test-utils';
import { unnnicTextArea } from '@weni/unnnic-system';
import FormTabContentFooter from '@/components/whatsAppTemplates/FormTabContentFooter.vue';
import i18n from '@/utils/plugins/i18n';

const localVue = createLocalVue();
localVue.use(Vuex);

const mountComponent = ({ footer = '' } = {}) => {
  const getters = {
    templateTranslationCurrentForm: jest.fn(() => {
      return { footer };
    }),
  };

  const store = new Vuex.Store({
    modules: {
      WhatsApp: {
        namespaced: true,
        getters,
      },
    },
  });

  const wrapper = mount(FormTabContentFooter, {
    localVue,
    store,
    i18n,
  });

  return { wrapper, getters };
};

describe('components/whatsAppTemplates/FormTabContentFooter.vue', () => {
  it('should be rendered properly', () => {
    const { wrapper } = mountComponent();
    expect(wrapper).toMatchSnapshot();
  });

  it('should emit input-change on header text change', async () => {
    const { wrapper } = mountComponent();
    const inputComponent = wrapper.findComponent(unnnicTextArea);

    inputComponent.vm.$emit('input', 'text123');
    await wrapper.vm.$nextTick();

    const event = wrapper.emitted('input-change');

    expect(event).toBeTruthy();
    expect(event.length).toBe(1);
    expect(event[0]).toEqual([
      {
        fieldName: 'footer',
        fieldValue: 'text123',
      },
    ]);
  });
});
