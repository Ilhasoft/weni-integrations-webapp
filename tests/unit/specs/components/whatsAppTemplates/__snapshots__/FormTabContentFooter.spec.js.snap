// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/whatsAppTemplates/FormTabContentFooter.vue should be rendered properly 1`] = `
<div class="form-tab-content-footer"><span class="form-tab-content-footer__title">
    Footer Text
  </span>
  <div data-v-24fcdb87="" class="form-tab-content-footer__input unnnic-text-area md normal">
    <!----><textarea data-v-24fcdb87="" maxlength="60"></textarea>
    <div data-v-24fcdb87="" class="helper"> 0/60 </div>
    <!---->
  </div>
</div>
`;
